owncloudApp/src/androidTest/java/com/owncloud/android/settings/camerauploads/OCSettingsCameraUploadsTest.kt

```diff
@@ -0,0 +1,255 @@
+            mPrefCameraPictureUploads.setChecked(false)
+            mPrefCameraVideoUploads.setChecked(false)
+        })
+
+    }
+
+    @Test
+    fun checkTitle() {
+        //Asserts
+        onView(withText(R.string.actionbar_settings)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun pictureUploadsView() {
+        //Asserts
+        onView(withText(R.string.prefs_camera_picture_upload)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_picture_upload_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_picture_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_picture_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun videoUploadsView() {
+        //Asserts
+        onView(withText(R.string.prefs_camera_video_upload)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_video_upload_summary)).check(matches(isDisplayed()))
+        onView(withText(R.string.prefs_camera_video_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_video_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun optionsCameraFolderBehaviour() {
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .check(doesNotExist())
+        onView(withText(R.string.prefs_camera_upload_behaviour_title)).check(doesNotExist())
+        onView(withText(R.string.pref_behaviour_entries_keep_file)).check(doesNotExist())
+    }
+
+    @Test
+    fun enablePictureUploads() {
+        onView(withText(R.string.prefs_camera_picture_upload)).perform(click());
+        onView(withText(android.R.string.ok)).perform(click())
+        //Asserts
+        assertTrue(mPrefCameraPictureUploads.isChecked)
+        onView(withText(R.string.prefs_camera_picture_upload_path_title)).check(matches(isDisplayed()))
+        onView(withText(R.string.camera_picture_upload_on_wifi)).check(matches(isDisplayed()))
+        //Reset suboptions
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun enableVideoUploads() {
+        onView(withText(R.string.prefs_camera_video_upload)).perform(click());
+        onView(withText(android.R.string.ok)).perform(click())
+        //Asserts
+        assertTrue(mPrefCameraVideoUploads.isChecked)
+        onView(withText(R.string.prefs_camera_video_upload_path_title)).check(matches(isDisplayed()))
+        onView(withText(R.string.camera_video_upload_on_wifi)).check(matches(isDisplayed()))
+        //Reset suboptions
+        removeVideoSubOptions()
+    }
+
+    @Test
+    fun disablePictureUploadsAccept() {
+        enablePictureSubOptions()
+        onView(withText(R.string.prefs_camera_picture_upload)).perform(click());
```

Comment:
Use removePictureSubOptions() here? And maybe you do not need to call enablePictureSubOptions() just before?

Reply:
Every test in the suite starts with all options disabled, as the `setUp()` method. The test try to check the status after disabling the preference. Before, it must be enabled.